arches/settings_utils.py

```diff
@@ -1,24 +1,67 @@
-import importlib
 import json
 import os
 import site
 import sys
+from contextlib import contextmanager
 
-from pathlib import Path
+import django
+from django.apps import apps, AppConfig
+from django.conf import global_settings, settings
+
+
+@contextmanager
+def move_to_end_of_sys_path(*paths):
+    _orig_sys_path = sys.path[:]
+    for path in paths:
+        if path in sys.path:
+            sys.path.remove(path)
+            sys.path.append(path)
+    try:
+        yield
+    finally:
+        sys.path = _orig_sys_path
+
+
+def arches_applications():
+    from arches import settings as core_settings
+
+    return [
+        mod.name for mod in arches_applications_modules(core_settings.INSTALLED_APPS)
+    ]
+
+
+def arches_applications_modules(installed_apps):
+    from arches import settings as core_settings
+
+    arches_applications_modules = []
+
+    for app in installed_apps or []:
+        if app in core_settings.INSTALLED_APPS:
+            continue
```

Comment:
I'm unsure if this logic block is good for future-proofing. I believe in the future Arches core will used Arches applications